src/dto/error-suppressions-payload.dto.ts

```diff
@@ -1,34 +1,142 @@
 import {
   IsBoolean,
-  IsDate,
   IsDateString,
+  IsNotEmpty,
   IsNumber,
+  IsOptional,
   IsString,
+  ValidationArguments,
 } from 'class-validator';
+import { CheckCatalogService } from '@us-epa-camd/easey-common/check-catalog';
+import { ErrorMessages } from '@us-epa-camd/easey-common/constants';
+import { IsValidCode, IsValidDate } from '@us-epa-camd/easey-common/pipes';
+import { Plant } from '../entities/plant.entity';
+import { EsReasonCode } from '../entities/es-reason-code.entity';
+import { CheckCatalogResult } from '../entities/check-catalog-result.entity';
+import { SeverityCode } from '../entities/severity-code.entity';
+import { EsMatchDataTypeCode } from '../entities/es-match-data-type-code.entity';
+import { EsMatchTimeTypeCode } from '../entities/es-match-time-type-code.entity';
+
+const msgA = `You reported an invalid [value] of [property]`;
 
 export class ErrorSuppressionsPayloadDTO {
+  @IsNotEmpty({
+    message: () => {
+      return ErrorMessages.RequiredProperty();
+    },
+  })
   @IsNumber()
+  @IsValidCode(CheckCatalogResult, {
+    message: (args: ValidationArguments) => {
+      return CheckCatalogService.formatMessage(msgA, {
+        property: args.property,
+        value: args.value,
+      });
+    },
+  })
   checkCatalogResultId: number;
+
+  @IsNotEmpty({
+    message: () => {
+      return ErrorMessages.RequiredProperty();
+    },
+  })
   @IsString()
+  @IsValidCode(SeverityCode, {
+    message: (args: ValidationArguments) => {
+      return CheckCatalogService.formatMessage(msgA, {
+        property: args.property,
+        value: args.value,
+      });
+    },
+  })
   severityCode: string;
+
   @IsNumber()
-  facilityId: number;
+  @IsValidCode(Plant, {
+    message: (args: ValidationArguments) => {
+      return `The ${args.property} is not valid. Refer to the list of available facilityRecordIds for valid values '/facilities-mgmt/facilities'`;
+    },
+  })
+  @IsOptional()
+  facilityId?: number;
+
   @IsString()
-  locations: string;
+  @IsOptional()
+  locations?: string;
+
   @IsString()
-  matchDataTypeCode: string;
+  @IsValidCode(EsMatchDataTypeCode, {
+    message: (args: ValidationArguments) => {
+      return CheckCatalogService.formatMessage(msgA, {
+        property: args.property,
+        value: args.value,
+      });
+    },
+  })
+  @IsOptional()
+  matchDataTypeCode?: string;
+
   @IsString()
-  matchDataValue: string;
+  @IsOptional()
+  matchDataValue?: string;
+
+  @IsNotEmpty({
```

Comment:
Since it's optional the IsNotEmpty needs to be removed

Reply:
updated